odoo/tools/misc.py

```diff
@@ -1372,36 +1373,42 @@ def babel_locale_parse(lang_code):
         except:
             return babel.Locale.parse("en_US")
 
-def formatLang(env, value, digits=None, grouping=True, monetary=False, dp=False, currency_obj=False):
+def formatLang(env, value, digits=2, grouping=True, monetary=False, dp=None, currency_obj=None, rounding_mode=ROUND_HALF_EVEN, amount_rounding='decimals'):
     """
         Assuming 'Account' decimal.precision=3:
             formatLang(value) -> digits=2 (default)
             formatLang(value, digits=4) -> digits=4
             formatLang(value, dp='Account') -> digits=3
             formatLang(value, digits=5, dp='Account') -> digits=5
     """
+    # We don't want to return 0
+    if isinstance(value, str) and not value:
+        return ''
 
-    if digits is None:
-        digits = DEFAULT_DIGITS = 2
+    if amount_rounding == 'decimals':
         if dp:
-            decimal_precision_obj = env['decimal.precision']
-            digits = decimal_precision_obj.precision_get(dp)
+            digits = env['decimal.precision'].precision_get(dp)
         elif currency_obj:
             digits = currency_obj.decimal_places
+    else:
+        digits = 0
 
-    if isinstance(value, str) and not value:
-        return ''
-
-    lang_obj = get_lang(env)
+    if amount_rounding == 'thousands':
+        value = value / 10**3
+    elif amount_rounding == 'lakhs':
+        value = value / 10**5
+    elif amount_rounding == 'millions':
+        value = value / 10**6
 
-    res = lang_obj.format('%.' + str(digits) + 'f', value, grouping=grouping, monetary=monetary)
+    rounded_value = Decimal(str(value)).quantize(Decimal(10) ** -digits, rounding_mode)
+    formatted_value = get_lang(env).format('%.' + str(digits) + 'f', rounded_value, grouping=grouping, monetary=monetary)
```

Comment:
```suggestion
    formatted_value = get_lang(env).format(f'%.{digits}f', rounded_value, grouping=grouping, monetary=monetary)
```
More pythonic :wink: